README.md

```diff
@@ -1,189 +1,248 @@
-```
-private_key
-sign_cert
+```shell
+c2patool -c '{"assertions": [{"label": "org.contentauth.test", "data": {"name": "Jane Doe"}}]}'
 ```
-If you are using a signing algorithm other than the default ps256, you will need to specify it in
-```alg```
-Which can be set to one of [ ps256 | ps384 | ps512 | es256 | es384 | es512 | ed25519] and
-must be compatible with values of private key  and sign cert.
+ 
+### Adding a manifest to a file
+
+#### Setup
+
+##### Creating and using an X.509 certificate
+
+Before you can add a manifest, you need to create an X.509 certificate. You can specify the path to the cert files in the following configuration fields:
+
+- `private_key`
+- `sign_cert`
 
-The key and cert can also be placed in the environment variables C2PA_PRIVATE_KEY and C2PA_PUB_CERT  
-These two variable are used to set the private key and public certificates.  For example to sign with es256 signatures
-using the content of a private key file and certificate file:
+If you are using a signing algorithm other than the default `ps256`, you will need to specify it in `alg`, which can be set to one of the following:
 
-```set C2PA_PRIVATE_KEY=$(cat my_es256_private_key)```
-```set C2PA_PUB_CERT=$(cat my_es256_certs)```
+- `ps256`
+- `ps384`
+- `ps512`
+- `es256`
+- `es384`
+- `es512`
+- `ed25519`
 
-The both private key and sign cert should be in PEM format.  The sign cert should contain a certificate
-chain PEMs starting for the end-entity certificate used to sign the claim ending with intermediate certificate
-before the root CA certificate.  See ```sample`` folder for example certificates.
+The specified algorithm must be compatible with values of `private_key` and `sign_cert`.
 
-To create your own temporary files for testing you can execute the following command
+The key and cert can also be placed in the environment variables `C2PA_PRIVATE_KEY` and `C2PA_PUB_CERT`. These two variables are used to set the private key and public certificates. For example, to sign with es256 signatures using the content of a private key file and certificate file, you would run:
 
 ```shell
-sudo openssl req -new -newkey rsa:4096 -sigopt rsa_padding_mode:pss -days 180 -extensions v3_ca -addext "keyUsage = digitalSignature" -addext "extendedKeyUsage = emailProtection" -nodes -x509 -keyout private.key -out certs.pem -sha256
+set C2PA_PRIVATE_KEY=$(cat my_es256_private_key)
+set C2PA_PUB_CERT=$(cat my_es256_certs)
+```
+
+Both the `private_key` and `sign_cert` should be in PEM format. The `sign_cert` should contain a PEM certificate chain starting for the end-entity certificate used to sign the claim ending with the intermediate certificate before the root CA certificate. See the ["sample" folder](https://github.com/contentauth/c2patool/tree/main/sample) for example certificates.
+
+To create your own temporary files for testing, you can execute the following command:
+
+```shell
+sudo openssl req -new -newkey rsa:4096 
```

Comment:
@gpeacock Do we need `sudo` here? I don't see anything that requires it.

Reply:
maybe we don't need it in this context, i'm not sure